File: remote/src/main/java/com/kelsos/mbrc/data/UserAction.java
package com.kelsos.mbrc.data;

public class UserAction {
  private String context;
  private Object data;

  public UserAction(String context, Object data) {
    this.context = context;
    this.data = data;
  }

  public String getContext() {
    return context;
  }

  public Object getData() {
    return data;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/events/UserAction.kt
package com.kelsos.mbrc.events

import com.kelsos.mbrc.networking.protocol.Protocol
import com.kelsos.mbrc.networking.protocol.Protocol.Context

class UserAction(@Context val context: String, val data: Any) {
  companion object {

    fun create(@Context context: String): UserAction {
      return UserAction(context, true)
    }

    fun create(@Context context: String, data: Any): UserAction {
      return UserAction(context, data)
    }

    fun toggle(@Context context: String): UserAction {
      return UserAction(context, Protocol.TOGGLE)
    }
  }
}
